src/showingpreviously/cinemas/everyman.py

```diff
@@ -0,0 +1,89 @@
+import json
+
+from datetime import datetime
+from typing import Tuple
+import showingpreviously.requests as requests
+from showingpreviously.model import ChainArchiver, CinemaArchiverException, Chain, Cinema, Screen, Film, Showing
+from showingpreviously.consts import UK_TIMEZONE
+
+
+CINEMAS_API_URL = 'https://www.everymancinema.com/cinemas'
+SHOWINGS_API_URL = 'https://movieeverymanapi.peachdigital.com/movies/13/{cinema_id}'
+
+CHAIN = Chain('Everyman Cinemas')
+
+
+def get_response(url: str) -> requests.Response:
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise CinemaArchiverException(f'Got status code {r.status_code} when fetching URL {url}')
+    return r
+
+
+def get_cinemas_as_dict() -> dict[str, Cinema]:
+    r = get_response(CINEMAS_API_URL)
+    try:
+        cinemas_data = r.json()
+    except json.JSONDecodeError:
+        raise CinemaArchiverException(f'Error decoding JSON data from URL {CINEMAS_API_URL}')
+    cinemas = {}
+    for cinema in cinemas_data:
+        id = cinema['CinemaId']
+        name = cinema['CinemaName']
+        cinemas[id] = Cinema(name, UK_TIMEZONE)
+    return cinemas
+
+
+def get_attributes_and_title(film_title:str) -> Tuple[str, dict[str, any]]:
+    lives = ['National Theatre Live:','ROH Live:', 'ROH Encore:', 'Met Opera Encore:', 'Met Opera Live:', ]
+    removals = ["Members' Preview:", '(Encore)', '+ Q&A', '+ Live Q&A', 'UK Jewish Film:']
+    attributes = {'format':[]}
+
+    if '(35mm)' in film_title:
+        film_title = film_title.replace('(35mm)', '')
+        attributes['format'].append('35mm')
+
+    for live in lives:
+        if live in film_title:
+            film_title = film_title.replace(live, '')
+            if 'live' not in attributes['format']:
+                attributes['format'].append('live')
+
+    for removal in removals:
+        if removal in film_title:
+            film_title = film_title.replace(removal, '')
+
+    return film_title, attributes
+
+
+def get_showings_date(cinema_id: str, cinema: Cinema) -> [Showing]:
+    url = SHOWINGS_API_URL.format(cinema_id=cinema_id)
+    r = get_response(url)
+    try:
+        showings_data = r.json()
+    except json.JSONDecodeError:
+        raise CinemaArchiverException(f'Error decoding JSON data from URL {url}')
+
+    showings = []
+    for film_data in showings_data:
+        film_title = film_data['Title']
+        film_year = film_data['ReleaseDate'][:4]
```

Comment:
This needs to be validated in case the API changes and the archiver starts adding in garbage data for the release year. However, my preference would be to do the validation in the model.